### PR TITLE
Enable Travis CI, simplified Debian packaging

### PR DESCRIPTION
**Pull request by lslezak**
# Travis CI

- Builds snapper using Docker images
- Builds several distros in parallel - Tumbleweed, the latest Leap (currently 42.2), Fedora 25, Debian 8 and Ubuntu 16.10
- Builds the binary packages (DEB or RPM), runs the tests during builds, installs the built binary packages and runs simple `snapper --version` to make sure it really works
- Documented in the `README.Travis.md` file

# Debian Packaging

- Simplified the `debian.rules` file
- Upgraded compatibility level to 7 (with level 5 the native Debian package builds failed)
- Allow parallel builds (much faster builds in OBS and Travis)
- Move the files in `*.install` files instead of `debian.rules` file workarounds
- Create the DSC file also for Ubuntu 16.10 to build it in OBS

# Tests

- The Travis builds pass, all builds for all configured distros pass, check the results in this PR
- I have tested the updated Debian packaging in my [OBS repo fork](https://build.opensuse.org/package/show/home:lslezak:branches:filesystems:snapper/snapper) - the package builds for all distributions even for the newly enabled Ubuntu 16.10

### CODE REVIEW

File: README.Travis.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+## Running the Build Locally
+
+- [Install and start Docker](https://docs.docker.com/engine/installation/linux/)
+- Run (as `root`) the same commands as in the `.travis.yml` file, that means:
+- First build the docker image locally, e.g. `docker build -f
+  Dockerfile.tumbleweed -t snapper-devel .`, the Docker image automatically
+  includes also the copy of the current Snapper sources.
+- Then run the build: `docker run -it --rm snapper-devel ./.travis.tumbleweed.sh`
+  (The `--rm` will cleanup the new layer created by the build, if want to
```
Comment (aschnell) [inline README.Travis.md]: "if you want"

File: .travis.fedora.sh [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+# "osc build" is too resource hungry (builds a complete chroot from scratch).
+# Moreover it does not work in a Docker container (it fails when trying to mount
+# /proc and /sys in the chroot).
+mkdir -p /root/rpmbuild/SOURCES
+cp package/* /root/rpmbuild/SOURCES
+rpmbuild -bb -D "fedora_version 25" -D "jobs `nproc`" package/*.spec
+
+# test the %pre/%post scripts by installing/updating/removing the built packages
+# ignore the dependencies to make the test easier, as a smoke test it's good enough
+rpm -iv --force --nodeps /root/rpmbuild/RPMS/**/*.rpm
```
Comment (aschnell) [inline .travis.fedora.sh]: Why twice "*"?

Comment (lslezak) [inline .travis.fedora.sh]: Oh, in Ruby it means recursive search, but that does not work in shell (unless the `globstar` feature is enabled). I'll fix that...

**Comment by ancorgs** [comment]
I'm far from understanding everything in detail, but it LGTM.

**Pull request merged**